lib/generic/ParamController.hpp

```diff
@@ -24,4 +24,7 @@ class ParamController {
   virtual uint8_t GetRawParam(Param param) = 0;
 
   virtual int16_t GetScaledParam(Param param, int16_t min, int16_t max);
+
+  // Perform periodic work
+  virtual void Run(){};
```

Comment:
I would expect a function named "`Run`" to be a blocking function that just goes -- the equivalent of `loop` right now. Maybe `Step` would be a better name?